src/gdbstub_impl/ext/single_register_access.rs

```diff
@@ -13,17 +13,40 @@ impl<T: Target, C: Connection> GdbStubImpl<T, C> {
     ) -> Result<HandlerStatus, Error<T::Error, C::Error>> {
         let handler_status = match command {
             SingleRegisterAccess::p(p) => {
-                let mut dst = [0u8; 32]; // enough for 256-bit registers
                 let reg = <T::Arch as Arch>::RegId::from_raw_id(p.reg_id);
                 let (reg_id, reg_size) = match reg {
                     // empty packet indicates unrecognized query
                     None => return Ok(HandlerStatus::Handled),
                     Some(v) => v,
                 };
-                let dst = &mut dst[0..reg_size];
-                ops.read_register(id, reg_id, dst).handle_error()?;
 
-                res.write_hex_buf(dst)?;
+                let mut n = 0usize;
+                let mut err = Ok(());
+
+                ops.read_register(
+                    id,
+                    reg_id,
+                    SendRegisterOutput::new(&mut |buf| {
+                        if err.is_ok() {
+                            // If the register has a known size and read_register attempts
+                            // to send more bytes than are present in the register,
+                            // error out and stop sending data.
+                            n += buf.len();
```

Comment:
move this line + the set error line into the `if let Some` branch. If `from_raw_id` is inlined + always returns `Some(usize, None)`, then the entire contents of the callback can be DCEd

Reply:
👀 which error line? I can't see how to move the `err = res.write_hex_buf ...` line.

Reply:
ah, uhh, nevermind 😅
coffee hasn't fully kicked in yet